src/libs/actions/Policy.js

```diff
@@ -112,6 +112,15 @@ function getPolicyList() {
         });
 }
 
+/**
+ * Does the user have any (free policy) workspaces?
+ *
+ * @returns {Boolean}
+ */
+function hasWorkspaces() {
+    return _.some(allPolicies, policy => policy && policy.type === CONST.POLICY.TYPE.FREE && policy.role === CONST.POLICY.ROLE.ADMIN);
```

Comment:
This might work, but wanted to point out something about "Onyx" here that might be a "gotcha" or just plain confusing.

Since we are relying on a local collection of policies here if we try to use this method in a component then the component won't re-render with the new result. To do that, we'd also need to subscribe to the `policies` object in that component. But even then, there's no guarantee the local policy object here would be updated in time for this function to be accurate because there's no guarantee that the local `Onyx.connect()` callback will run before our component re-renders.

So.... Does that matter? Do we need it to work another way? It probably depends on the situation. 

However, by making this function pure and passing in the list of policies as an argument we could make it more useful e.g.

```js
function isAdminOfFreePolicy(policies) {
    return _.some(policies, policy => policy 
        && policy.type === CONST.POLICY.TYPE.FREE 
        && policy.role === CONST.POLICY.ROLE.ADMIN);
}
```

That way this function does not need to worry if the list of policies it has are "up to date" instead given any number of policies we will return if one is a free policy.

Then for example we could do something like

```jsx
import {isAdminOfFreePolicy} from './actions/Policy';

const MyComponent = (props) => {
    if (isAdminOfFreePolicy(props.allPolicies)) {
        return <OneView />
    }
    
    return <SomeOtherView />
};

export default withOnyx({
    allPolicies: {key: ONYXKEYS.COLLECTION.POLICY},
})(MyComponent);
```

By subscribing to Onyx key in the component we we guarantee the list of policies is up to date and that the component will re-render when any policies change so it can stay up to date.

Reply:
Nice catch there, didn't know this could be a problem. I see other functions in `Policy.js` using `allPolicies` variable, can they potentially run into the same problem? Or they don't have this problem because they are function that won't be called during `rendering`? Example:

https://github.com/Expensify/App/blob/4e9539e09346ede511c8a92172cfb93c7d00d9bb/src/libs/actions/Policy.js#L64-L78

Reply:
>I see other functions in Policy.js using allPolicies variable, can they potentially run into the same problem?

They don't have the same exact problem I'm describing because we don't need these methods to be accurate or run inside a React component.

That said, they _could_ have a similar problem if we assume that the `allPolicies` we are referencing has actually been updated when it has not. It really depends on the case and how things are being used. IMO very confusing and a design about Onyx that needs to be discussed more before we invest more into it.

Reply:
Ah, that all makes a lot of sense! I'll update it 👍